runtime/jcl/common/jcldefine.c

```diff
@@ -211,18 +212,25 @@ defineClassCommon(JNIEnv *env, jobject classLoaderObject,
 	}
 
 done:
-	if ((clazz == NULL) && (currentThread->currentException == NULL)) {
-		/* should not get here -- throw the default exception just in case */
-		vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGCLASSFORMATERROR, NULL);
-	}
-			
-	result = vmFuncs->j9jni_createLocalRef(env, J9VM_J9CLASS_TO_HEAPCLASS(clazz));
+	if (clazz == NULL) {
+		if (J9_ARE_ANY_BITS_SET(options, J9_FINDCLASS_FLAG_NAME_IS_INVALID)) {
+			/*
+			 * The caller signalled that the name is invalid. Leave the result NULL and
+			 * clear any pending exception; the caller will throw NoClassDefFoundError.
+			 */
+			currentThread->currentException = NULL;
```

Comment:
This should clear the privateFlags bit as well:

https://github.com/eclipse/openj9/blob/dde0f75a7bf23f7203499db658c1561c26b87e4c/runtime/oti/VMHelpers.hpp#L637

Reply:
Thanks; fixed.